### PR TITLE
Implement financial index API

### PR DESCRIPTION
**Pull request by jaredkosanovic**
This API is to validate financial account index codes, which are codes that are used for purchasing so that the appropriate unit/department is charged for a purchase.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -65,44 +65,42 @@ adminAppRouter.get('/', async (req, res) => {
...
+      // If there's additional parameters in the query object, oracle will complain
+      // because it doesn't recognize them. Thus, we only add the ones we need,
+      // and the ones that are defined.
+      // TODO: There's got to be a less ugly way to do this.
+      const params = {};
+      if (q) {
+        params.q = q;
+      }
+      if (organizationCode) {
+        params.organizationCode = organizationCode;
```
Comment (tsoliangwu0130) [inline app.js]: Good catch, I should also add some params cleaner function to the skeleton. You could probably use [lodash](https://lodash.com/docs/4.17.10) to shorten the code:

```suggestion
      _.forEach(req.query, (value, key) => {
        if (value !== '') params[key] = value;
      });
```

you will need to require lodash at the top of the file:

```js
const _ = require('lodash');
```

Comment (jaredkosanovic) [reply app.js]: Mentioned this in hipchat, but I still need a way to only include parameters that I'm expecting (q and organizationCode). For now, I'll continue explicitly getting the expected parameters. In the future, this should be inherited from the swagger definition.

File: app.js [hunk 2]
```diff
@@ -65,44 +68,77 @@ adminAppRouter.get('/', async (req, res) => {
...
     }
+  } catch (err) {
+    errorHandler(res, err);
+  }
+});
 
-    const result = await db.getApis(req.query);
-    return res.send(result);
+/**
+ * @summary Get API by unique ID
```
Comment (tsoliangwu0130) [inline app.js]: This description should be changed.

File: tests/unit/serializers.test.js [hunk 3]
```diff
@@ -3,48 +3,18 @@ const camelCase = require('camelcase');
...
 
 describe('Test JSON API serializer', () => {
-  const jsonapi = apiResourceSerializer(rows, 'exampleUri');
+  const jsonapi = accountIndexSerializer(rows, 'exampleUri');
   it('keys should be camelCase', (done) => {
     const newKeys = _.keys(jsonapi.data[0].attributes);
     _.forEach(newKeys, key => assert.equal(key, camelCase(key)));
     done();
   });
 });
```
Comment (tsoliangwu0130) [inline tests/unit/serializers.test.js]: This test case will call uri-builder which will load the configuration file during the building process:

```
[16:31:23] Starting 'test'...
WARNING: No configurations found in configuration directory:/usr/src/financial-index-api/config
WARNING: To disable this warning set SUPPRESS_NO_CONFIG_WARNING in the environment.
/usr/src/financial-index-api/node_modules/config/lib/config.js:203
    throw new Error('Configuration property "' + property + '" is not defined');
```

This issue was fixed in this PR: https://github.com/osu-mist/express-api-skeleton/pull/7
In addition, we don't want to remove paginator test cases from the unit tests as well.

File: app.js (hunk 2 above)
Comment (jaredkosanovic) [reply app.js]: Ah, good catch. There's a lot of things that need to be changed from the skeleton.

File: serializers/uri-builder.js [hunk 4]
```diff
@@ -4,7 +4,17 @@ const fs = require('fs');
 const yaml = require('js-yaml');
 const url = require('url');
 
-const { protocol, hostname } = config.get('server');
+// Configuration file won't exist until deployment.
+// Use dummy values if we can't load the configuration file.
+function getConfigValues() {
```
Comment (tsoliangwu0130) [inline serializers/uri-builder.js]: Good idea, but can you use arrow function here as well for the consistency? Also, I think in ECMAScript 6 or above, we should use arrow function as much as we can. https://stackoverflow.com/questions/22939130/when-should-i-use-arrow-functions-in-ecmascript-6

**Pull request merged**